app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +280,141 @@ angular.module('confRegistrationWebApp')
+            // A. If spouse is not included in that registration, then show another separate panel on the confirmation page,
+            //    indicating that their spouse has already registered and would they like to add their registration to their spouses?
+            //    . Call endpoint to copy registrant from one registration to another
+            //    . (possibly) Call updateRegistration
+            if (!spouseIncludedOnCurrentRegistration) {
+              $scope.spouseSectionVisible = true;
+            }
+            // B. Otherwise, indicate that they have already been included on their spouses registration. And possibly let them register anyway if they must.
+            else {
+              $scope.spouseSectionVisible = false;
+            }
+          }
+        }).catch(function(response){
+        console.log('Failed to get all registered users for current conference. Status = ' + response.status);
+      });
+    };
+
+    //Click event for 'Include Spouse' button
+    $scope.includeSpouseInRegistration = function () {
+
+      //Generate new local UUID used for registrantId
+      var _newRegistrantId = uuid();
+
+      //Payload for new spouse registration
+      var _newSpouseRegistration =
+        {
+          id: $scope.spouseRegistration.id,
+          conferenceId: $scope.currentRegistration.conferenceId,
+          registrants: [{
+            id: _newRegistrantId,
+            registrationId: $scope.spouseRegistration.id,
+            registrantTypeId: $scope.currentRegistration.registrants[0].registrantTypeId,
+            answers: []
+          }]
+        };
+
+      //Get all answers
+      var _answers = $scope.currentRegistration.registrants[0].answers.map(function(answer) {
+        return _.assign({}, answer, {
+          id: uuid(),
+          registrantId: _newRegistrantId
+        });
+      });
+
+      //Payload to add new registrants
+      var _newRegistrants =
+      {
+        id: _newRegistrantId,
+        registrationId: $scope.spouseRegistration.id,
+        registrantTypeId: $scope.currentRegistration.registrants[0].registrantTypeId,
+        answers: _answers,
+        firstName: $scope.currentRegistration.registrants[0].firstName,
+        lastName: $scope.currentRegistration.registrants[0].lastName,
+        email: $scope.currentRegistration.registrants[0].email
+      };
+
+      //Add new registration
+      //@Path("/registrations/{registrationId}")
+      $http.put('/registrations/' + $scope.spouseRegistration.id, _newSpouseRegistration)
+        .then(function () {
+          //Add registrants to new registration
+          //@Path("/registrants/{registrantId}")
+          return $http.put('/registrants/' + _newRegistrantId, _newRegistrants);
+        }).then(function () {
+          //Delete existing registration
+          //@Path("/registrants/{registrantId}")
+          return $http.delete('/registrants/' + $scope.currentRegistration.id, $scope.currentRegistration);
```

Comment:
I just realized that here we are deleting the registrant when we actually need to delete the entire registration. Also, we only need to call `$http.delete` with the URL argument; you can remove the second argument.